components/feature/intent/src/main/java/mozilla/components/feature/intent/IntentProcessor.kt

```diff
@@ -51,7 +51,7 @@ class IntentProcessor(
                     val displayMetrics = context.resources.displayMetrics
                     this.customTabConfig = CustomTabConfig.createFromIntent(safeIntent, displayMetrics)
                 }
-                sessionManager.add(session)
+                sessionManager.add(session, selected = true)
```

Comment:
While this may work in this case this solution breaks a "hidden" contract: The selected session is the "selected tab" in the browser and therefore **never** a custom tab. We failed to make this an explicit API in `SessionManager` - hopefully we can do that better in `browser-state` (e.g. we want to avoid mixing "normal" and custom tab sessions in the same list again).

Reply:
(The side effect of this patch will be that the custom tab will show up as the active tab in the browser too. And if we will have two activities (brower & custom tab) render the same session GV will even crash).